docs/tables.md

```diff
@@ -0,0 +1,458 @@
+* The `table_attrs` parameter, as a shorthand for updating the Zarr attributes of the table group after its creation.
+
+Here is an example of how to use `write_table`:
+```python
+import numpy as np
+import zarr
+import anndata as ad
+from fractal_tasks_core.lib_write import write_table
+
+table = ad.AnnData(X=np.ones((10, 10)))  # Generate a dummy `AnnData` object
+image_group = zarr.open_group("/tmp/image.zarr")
+table_name = "MyTable"
+table_attrs = {
+    "type": "ngff:region_table",
+    "region": {"path": "../labels/MyLabel"},
+    "instance_key": "label",
+}
+
+write_table(
+    image_group,
+    table_name,
+    table,
+    overwrite=True,
+    table_attrs=table_attrs,
+)
+```
+After running this Python code snippet, the on-disk output  is as follows:
+```console
+$ tree /tmp/image.zarr/tables/                  # View folder structure
+/tmp/image.zarr/tables/
+└── MyTable
+    ├── layers
+    ├── obs
+    │   └── _index
+    │       └── 0
+    ├── obsm
+    ├── obsp
+    ├── uns
+    ├── var
+    │   └── _index
+    │       └── 0
+    ├── varm
+    ├── varp
+    └── X
+        └── 0.0
+
+12 directories, 3 files
+
+$ cat /tmp/image.zarr/tables/.zattrs            # View tables atributes
+{
+    "tables": [
+        "MyTable"
+    ]
+}
+
+$ cat /tmp/image.zarr/tables/MyTable/.zattrs    # View single-table attributes
+{
+    "encoding-type": "anndata",
+    "encoding-version": "0.1.0",
+    "fractal_table_version": "1",
+    "instance_key": "label",
+    "region": {
+        "path": "../labels/MyLabel"
+    },
+    "type": "ngff:region_table"
+}
+```
+
+## Outlook
+
+These specifications may evolve (especially based on the future NGFF updates),
+eventually leading to breaking changes in future versions.
+`fractal-tasks-core` will aim at mantaining backwards-compatibility with V1 for
+a reasonable amount of time.
+
+Here is an in-progress list of aspects that may be reviewed:
+
+* We aim at removing the use of hard-coded units from the column names (e.g.
+  `x_micrometer`), in favor of a more general definition of units.
+* The `z_micrometer` and `len_z_micrometer` columns are currently required in
+  all ROI tables, even when the ROIs actually define a two-dimensional XY
+  region; in that case, we set `z_micrometer=0` and `len_z_micrometer` is such
+  that the whole Z size is covered. In a future version, we may introduce more
+  flexibility and also accept ROI tables which only include X and Y axes, and
+  adapt the relevant tools so that they automatically expand these ROIs into
+  three-dimensions when appropriate.
+* We may re-evaluate whether `AnnData` tables are the most appropriate tool. For
+  the record, Zarr does not natively support storage of dataframes (see e.g.
+  https://github.com/zarr-developers/numcodecs/issues/452), which is one
+  aspect in favor of sticking with the `anndata` library.
```

Comment:
Our plan is to follow whatever serialised table specification becomes part of the OME-NGFF standard.